### ISSUE TITLE
Retrieve full collection on value change within a collection

### ISSUE THREAD

**Issue by axiomabsolute** [javers/javers#95]
__EDIT__: Sorry for the long issue.  Maybe skip to the last paragraph and see if that makes sense on its own, and come back to read all this if you need more details.

Is it possible to retrieve the full collection when a value change occurs on an item within the collection, instead of just reporting the change made to that particular item?  For example, say I have models like a following:

```java
class Address {
    @Id
    String fullAddress;
}
class Person {
    @Id String _id;
    String name;
    List<Address> addresses;
}
```

If an address is added or removed from `addresses`, a ListChange is produced in the diff, and if an address is edited within the person, a ValueChange will be produced in the diff.

Is there a way to report the full collection, before and after, when iterating over the changes in the diff?  I know I can retrieve the Property value from `PropertyChange`s in the diff:

```java
for(PropertyChang pc : diff.getChangesByType(PropertyChange.class)){
    String propertyName = pc.getProperty().getName();
    Object beforeVal = pc.getProperty().get(oldRecord);
    Object afterVal = pc.getProperty().get(newRecord);
}
```

This works for changes made to top-level properties, like `name`, but not for properties within collections, like `fullAddress` within an `Address` in `addresses`.  In the case of changes made to entities within collections, the property returned by `pc.getProperty()` is the property _on the embedded entity_, i.e. it would return `fullAddress` as the property name.  Is there some way to use the global id to get back to the top level property changed?

Basically, I want:

```java
Address a1 = new Address(){{ this.id = 1; this.fullAddress = "test1"; }};
Address a2 = new Address(){{ this.id = 1; this.fullAddress = "test2"; }};
Person p1 = new Person(){{ this.address = new ArrayList(){{ this.add(a1); }}; }};
Person p2 = new Person(){{ this.address = new ArrayList(){{ this.add(a2); }}; }};
javers.compare(p1, p2);
```

to report the entire value of the `address` field when any element inside the collection is changed.

Any clues?

**Comment by bartoszwalacik** [comment]
Hi,
I think I understand what do you mean. Well, you always can refer ro changed object (in this case Address) using
http://javers.org/javadoc_1.0.0/org/javers/core/diff/Change.html#getAffectedCdo--

we call it affected (changed) cdo (client's domain object) - Address in your case, did you tried this?

**Comment by bartoszwalacik** [comment]
hope my answers helps, if not, please reopen this issue

**Comment by ranjitc5** [comment]
Is there any way to provide alias name to entity property?

**Comment by bartoszwalacik** [comment]
no aliases for properties
